Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/extrude.h

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) 2018 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0+
+//
+//
+// Author(s)     : Sebastien Loriot, Maxime Gimeno
+
+
+#ifndef CGAL_EXTUR_POLYGON_MESH_H
+#define CGAL_EXTUR_POLYGON_MESH_H
+
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/Polygon_mesh_processing/orientation.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+#include <CGAL/boost/graph/named_function_params.h>
+#include <CGAL/Kernel_traits.h>
+#include <boost/unordered_map.hpp>
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+
+/**
+ * \ingroup ???
+ * Extrudes `input` into `output` following the direction given by `dir` and
+ * at a distance `d`.
+ * @tparam InputMesh a model of the concept `FaceListGraph`
+ * @tparam OutputMesh a model of the concept `FaceListGraph`
+ * @tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+ * @param input the closed triangulated `InputMesh` to extrude.
+ * @param output the `OutputMesh` containing the result of the extrusion.
+ * @param np an optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{vertex_point_map}
+ *    the property map that will contain the points associated to the vertices of `output`. 
+ * If this parameter is omitted, an internal property map for `CGAL::vertex_point_t` 
+ * should be available for the vertices of `output` \cgalParamEnd
+ * \cgalNamedParamsEnd
+ */
+template <class InputMesh,
+          class OutputMesh,
+          class NamedParameters>
+void extrude_mesh(const InputMesh& input, 
+                  OutputMesh& output, 
+                  #ifdef DOXYGEN_RUNNING
+                  Vector_3 dir,
+                  #else
+                  typename CGAL::Kernel_traits<
+                  typename boost::property_traits<
+                  typename GetVertexPointMap <
+                  OutputMesh, NamedParameters>::type>::
+                  value_type>::Kernel::Vector_3 dir, 
+                  #endif
+                  const double d, 
+                  const NamedParameters& np)
```

Comment:
I guess two set of named parameters are needed, one for the input and one for the output

Reply:
The one from the input was not used anywhere, so I removed it.